trie/utils/branches.py

```diff
@@ -0,0 +1,46 @@
+from trie.constants import (
+    BLANK_HASH,
+    KV_TYPE,
+    BRANCH_TYPE,
+    LEAF_TYPE,
+    BYTE_0,
+)
+from trie.utils.binaries import (
+    encode_to_bin,
+)
+from trie.utils.nodes import (
+    parse_node,
+)
+
+
+def if_branch_exist(db, node_hash, key_prefix):
```

Comment:
Is this intended to be a public API?  If so, it should be moved outside of `utils` as `utils` is intended to be purely private APIs

Reply:
yes it's intended to be a public API. I will move it to `trie/branches.py`.